web/src/pages/Users.tsx

```diff
@@ -1,59 +1,131 @@
-import { Table } from 'antd';
+import { Checkbox, Table } from 'antd';
 import type { ColumnsType } from 'antd/es/table';
 import { useEffect, useState } from 'react';
-import { getUsers } from '../services/UserService';
+import UserForm from '../components/UserForm';
 import { UserSchema } from '../schemas/User';
+import { getUsers, putUser } from '../services/UserService';
 import '../styles/userTData.css';
-import UserForm from '../components/UserForm';
 
 export default function Users() {
     const [users, setUsers] = useState<UserSchema[]>([]);
+    const [showDeactivated, setShowDeactivated] = useState(false);
+
     const requestUsers = () => {
         getUsers()
-        .then(usersResponse => {
-            setUsers(usersResponse);
-            console.log('users retornados do request:', usersResponse);
-        });
+            .then(usersResponse => {
+                setUsers(usersResponse);
+                console.log('users retornados do request:', usersResponse);
+            });
     }
+
     useEffect(() => {
         requestUsers()
     }, []);
 
+    const handleDeactivateUser = (user: UserSchema) => {
+        const confirmResult = window.confirm('Tem certeza de que deseja desativar este usuário?');
+        if (confirmResult) {
+            const updatedUserData = {
+                id: user.id,
+                name: user.name,
+                registration: user.registration,
+                userType: user.userType,
+                email: user.email,
+                password: user.password,
+                insertDate: user.insertDate,
+                expiredDate: user.expiredDate
+            };
+            putUser(user.id, updatedUserData, 1)
+                .then((updatedUser) => {
+                    if (updatedUser) {
+                        requestUsers();
+                    }
+                })
+                .catch((error) => {
+                    console.error('Erro ao desativar o usuário:', error);
+                });
+        }
+    };
+
+    const handleActivateUser = (user: UserSchema) => {
+        const confirmResult = window.confirm('Tem certeza de que deseja ativar este usuário?');
+        if (confirmResult) {
+            const updatedUserData = {
+                id: user.id,
+                name: user.name,
+                registration: user.registration,
+                userType: user.userType,
+                email: user.email,
+                password: user.password,
+                insertDate: user.insertDate,
+                expiredDate: user.expiredDate
+            };
+            putUser(user.id, updatedUserData, 2)
+                .then((updatedUser) => {
+                    if (updatedUser) {
+                        requestUsers();
+                    }
+                })
+                .catch((error) => {
+                    console.error('Erro ao ativar o usuário:', error);
+                });
+        }
+    };
```

Comment:
podemos utilizar a mesma função para fazer as duas operações, basta incluirmos uma boolean como parametro assim como na função "putUser".

também sugiro alterar o nome da função `putUser` para `updateUserActiveStatus` em um outro comentário por questões de legibilidade e entendimento do código.

além disso, não precisamos construir um segundo `UserSchema` para fazer essa alteração.

```suggestion
    const handleChangeUserActiveStatus = (userId: number, newActiveStatus: boolean) => {
        const operation = newActiveStatus ? 'ativar' : 'desativar';
        const confirmResult = window.confirm('Tem certeza de que deseja ${operation} este usuário?');
        if (confirmResult) {
            updateUserActiveStatus(userId, newActiveStatus)
                .then((updatedUser) => {
                    if (updatedUser) {
                        requestUsers();
                    }
                })
                .catch((error) => {
                    console.error('Erro ao ativar o usuário:', error);
                });
        }
    };
```